app/models/event.rb

```diff
@@ -55,6 +57,25 @@ def waitlist
     first_come_first_served - participants
   end
 
+  def can_participate?
+    first_come_first_served.count <= capacity
+  end
+
+  def cancel_participation!(user:)
+    participation = self.participations.find_by(user_id: user.id)
+    participation.destroy
+
+    return unless participation.enable
+
+    event = self
+    first_waiting_participation = first_waiting_participation(event)
```

Comment:
変数名とメソッド名を同じ行にあるのは避けたほうが可読性が良さそうです。

```suggestion
    繰り上がりの人 = first_waiting_participation(event)
```

という変数名に慣れば良さそうです

あるいは変数スコープが十分に小さければ単にparticipationでも構いません。

Reply:
`move_up_participation`という変数名に変更しましたが、命名に違和感ありますでしょうか？🤔